experiments/schema-annot/annotated/hotcrp-annotated.sql

```diff
@@ -0,0 +1,413 @@
+  -- KEY timeModifiedContact (timeModified,contactId),
+  FOREIGN KEY (paperId) REFERENCES Paper(paperId),
+  FOREIGN KEY (contactId) OWNED_BY ContactInfo(contactId)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE PaperConflict (
+  id int PRIMARY KEY NOT NULL,
+  paperId int NOT NULL,
+  contactId int NOT NULL,
+  conflictType int NOT NULL,
+  FOREIGN KEY (paperId) REFERENCES Paper(paperId)
+  -- see issue #172: ownership/edge in data ownership graph dependent on column value of conflictType,
+  -- which could be co-author, institutional, etc.
+  -- FOREIGN KEY (contactId) REFERENCES ContactInfo(contactId)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+-- Added this table to work around issue #172. Extracts co-author conflictType from PaperConflict
+CREATE TABLE PaperAuthors (
+  id int PRIMARY KEY NOT NULL,
+  paperId int NOT NULL,
+  contactId int NOT NULL,
+  FOREIGN KEY (paperId) REFERENCES Paper(paperId),
+  FOREIGN KEY (contactId) OWNED_BY ContactInfo(contactId)
+)
+
+CREATE TABLE PaperOption (
+  id int PRIMARY KEY NOT NULL,
+  paperId int NOT NULL,
+  optionId int NOT NULL,
+  value int NOT NULL,
+  data text,
+  dataOverflow text,
+  FOREIGN KEY (paperId) REFERENCES Paper(paperId)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+CREATE TABLE PaperReview (
+  paperId int NOT NULL,
+  reviewId int NOT NULL,
+  contactId int NOT NULL,
+  reviewToken int NOT NULL,
+  reviewType int NOT NULL,
+  reviewRound int NOT NULL,
+  requestedBy int NOT NULL,
+  timeRequested int NOT NULL,
+  timeRequestNotified int NOT NULL,
+  reviewBlind int NOT NULL,
+  reviewModified int NOT NULL,
+  reviewAuthorModified int,
+  reviewSubmitted int,
+  reviewNotified int,
+  reviewAuthorNotified int NOT NULL,
+  reviewAuthorSeen int,
+  reviewOrdinal int NOT NULL,
+  reviewViewScore int NOT NULL,
+  timeDisplayed int NOT NULL,
+  timeApprovalRequested int NOT NULL,
+  reviewEditVersion int NOT NULL,
+  reviewNeedsSubmit int NOT NULL,
+  reviewWordCount int,
+  reviewFormat int,
+  overAllMerit int NOT NULL,
+  reviewerQualification int NOT NULL,
+  novelty int NOT NULL,
+  technicalMerit int NOT NULL,
+  interestToCommunity int NOT NULL,
+  longevity int NOT NULL,
+  grammar int NOT NULL,
+  likelyPresentation int NOT NULL,
+  suitableForShort int NOT NULL,
+  potential int NOT NULL,
+  fixability int NOT NULL,
+  tfields text,
+  sfields text,
+  data text,
+  PRIMARY KEY (reviewId),
+  -- UNIQUE KEY reviewId (reviewId),
+  -- KEY contactId (contactId),
+  -- KEY reviewType (reviewType),
+  -- KEY reviewRound (reviewRound),
+  -- KEY requestedBy (requestedBy),
+  FOREIGN KEY (paperId) ACCESSED_BY Paper(paperId),
+  FOREIGN KEY (contactId) OWNED_BY ContactInfo(contactId),
+  FOREIGN KEY (requestedBy) ACCESSED_BY ContactInfo(contactId),
+  -- author should see review, but not who reviewed it
+  -- only thing author should see is content of review, and when submitted (dates)
+  ON GET contact_id ANON (review_token, reviewType, reviewRound, requestedBy, reviewNotified)
```

Comment:
This does not look right to me.
1. There is no contact_id column.
2. The contact_id is the reviewer right? We should be anonymizing for the authors of the paper.

Maybe you meant `ON GET paperId ANON (....)`? As in, if the authors of the paper request their data, do not show them private info about the review? (Also please make sure to anonymize the identity of the reviewer and who requested the review etc).

Reply:
You're right, that should be `paperId` not `contact_id`. Fixed in recent commit.